### PR TITLE
proto(engine): rename and remove some useless rpc

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref #7619

### What is changed and how it works?
1. rename and remove some useless rpc 

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/proto/master.proto [hunk 1]
```diff
@@ -121,11 +114,10 @@ message Selector {
 
 message HeartbeatRequest {
     string executor_id = 1;
-    int32 resource_usage = 2;
-    int32 status = 3;
+    int32 status = 2;
```
Comment (CharlesCheung96) [inline engine/proto/master.proto:117]: Should the status field be removed?

Comment (gozssky) [inline engine/proto/master.proto:117]: `status` could also be removed.

File: engine/proto/master.proto [hunk 2]
```diff
@@ -136,8 +128,7 @@ message Executor {
     // name is the readable name of the executor.
     string name = 2;
     string address = 3;
-    int64  capability = 4;
```
Comment (CharlesCheung96) [inline engine/proto/master.proto:139]: Should the capability field be removed?

File: engine/proto/master.proto (hunk 1 above)
Comment (gozssky) [reply engine/proto/master.proto:117]: +1

File: engine/proto/master.proto (hunk 2 above)
Comment (gozssky) [reply engine/proto/master.proto:139]: +1

File: engine/proto/master.proto (hunk 1 above)
Comment (amyangfei) [reply engine/proto/master.proto:117]: Better to remove it, it is not used now.

File: engine/proto/master.proto [hunk 3]
```diff
@@ -169,8 +160,8 @@ message ListMastersResponse {
 
 message ScheduleTaskRequest {
     string task_id = 1;
-    int64 cost = 2;
-    repeated ResourceKey resource_requirements = 3;
+    // resources required by the task.
+    repeated ResourceKey resources = 3;
     repeated Selector selectors = 4;
```
Comment (amyangfei) [inline engine/proto/master.proto:165]: repeated ResourceKey resources = 2;
    repeated Selector selectors = 3;

**Review by gozssky — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**